relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py

```diff
@@ -28,6 +30,46 @@
 
 
 # See global variables at the end of the file
+
+def makedirs(path):
+    if not os.path.isdir(path):
+        os.makedirs(path)
+
+def run(cmd, input=None, stdout=DEVNULL, stderr=DEVNULL, capture_output=False, shell=False, check=False):
+    logger.debug("Execute: %s"%cmd)
+    if capture_output:
+        stdout=subprocess.PIPE
+        stderr=subprocess.PIPE
+    try:
+        process = subprocess.run(
+            args=cmd,
+            input=input,
+            stdout=stdout,
+            shell=shell,
+            check=check
+        )
+        return (process.return_code, process.stdout, process.stderr)
+    except AttributeError:
```

Comment:
why does this error happen?

Reply:
the run command is a python3 function from the subprocess module. Since we are still supporting centos7 which does not officially support python3 we need to keep compatibility with python2.7.
I could use Popen everywhere but choose to go for the more standard way in python3 which is using the new high level run function. If for some reasons if fails to use it (does not exists or too old python3 using a different signature), it will fall back on the Popen interface.

Reply:
could you add a comment stating it is the python2 version, so that we can remove it after